### ISSUE TITLE
Bug: [no-unused-vars] `reportUsedIgnorePattern` false positives on enum members

### ISSUE THREAD

**Issue by bradzacher** [typescript-eslint/typescript-eslint#10116]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.5.4&showAST=es&fileType=.tsx&code=KYDwDg9gTgLgBMAdgVwLZwGIQnA3gKDjgH0BBAGkLgCFKBffIA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tidmUQAmtAG4BDaKgwBtcNhyJo0DtEgAaWXKxzskMRICSAc07REABVH58CpuigA9APpqN2mImLL8AVX4CjJuaW1tC2GPhwiK4QAL6uALqycTFAA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
export enum Foo {
  _A,
  B,
}
```

### ESLint Config

```javascript
module.exports = {
  "rules": {
    "@typescript-eslint/no-unused-vars": [
      "error",
      {
        "varsIgnorePattern": "^_",
        "reportUsedIgnorePattern": true
      }
    ]
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

no errors because the enum members aren't variables that should be checked by the rule

### Actual Result

```
'_A' is marked as ignored but is used. Used vars must not match /^_/u. 2:3 - 2:5
```

**Comment by abrahamguo** [comment]
I've opened a PR for this specific problem.

However, I just want to confirm — it looks like right now, we always consider enum members 'used', regardless of whether they are actually used or unused (that's why they were getting reported by `reportUsedIgnorePattern`). Is that correct, or should we be correctly determining whether each enum member is used/unused, which would also fix this problem?

**Comment by bradzacher** [comment]
Nah the current behaviour is correct.
Technically enum members are scoped variables cos you can do things like
```ts
enum Foo {
  A,
  B = A,
}
```
However for the purposes of `no-unused-vars` they obviously should never be considered unused cos there's no way to use them directly.

In a way an enum is kinda like a namespace where the members are exported variables:
```ts
namespace Foo {
  export const A = 1;
  export const B = A;
}
```

So yeah we should never report on them ever in any form within the rule.
